### PR TITLE
VACMS-6455: add audit tool for VAMC system service headings

### LINKED ISSUE #6455: Audit tool VAMC system service for headings and "care we provide" text
## User Story or Problem Statement

questions we want to answer

1. How many nodes (and which ones) have/have not the "Care we provide at [system]" in heading or otherwise?
1. How many nodes (and which ones) have other headings (list of nids and editors)

## Acceptance Criteria
- [ ] We have answers to the above questions available via the UI at /admin/content/audit/care-we-provide (with CSV export?)
- [ ] Audit can be accessed in the application by Administrator and Content Admin, no one else

## Implementation steps
<optional>

## Design principles

Which [CMS design principle](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/cms/cms-product-design-principles.md) is at play?
- [ ] Purpose-driven: Creates an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] Efficient: Removes distractions and create clear, straightforward paths to get the job done.
- [ ] Approachable: Offers friendly guidance over authoritative instruction.
- [ ] Consistent: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] Empowering: Provide clear information to help editors make decisions about their work.

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [ ] `Platform Team`
- [x] `Sitewide CMS Team`

### PR DESCRIPTION
**Pull request by JayDarnell**
## Description

Closes #6455 

## Testing done

Visual Testing

## Screenshots

Screen Shot 2021-10-20 at 11 24 00 AM

Filters applied

Screen Shot 2021-10-20 at 11 24 17 AM

CSV button is present

Screen Shot 2021-10-20 at 11 24 57 AM

Exported CSV file contains rows matching the applied filters

## QA steps

As user with either the administrator role or the content admin role

1. [x] Visit /admin/content/audit/vamc-system-service-audit and verify the page loads
2. [x] Filter the view by VAMC System Description of Service **contains** `<h3>Care we provide at</h3>`
3. [x] Verify the view refreshes and each item includes this heading
4. [x] Scroll to the bottom and click the CSV icon to download a CSV of your filtered results
5. [x] Verify that the CSV contains the same set of filtered items
6. [x] Reset the view filters and filter the view by VAMC System Description of Service **does not contain** `<h3>`
7. [x] Verify the view refreshes and none of the items contain an h3 tag
8. [x] Scroll to the bottom and click the CSV icon to download a CSV of your filtered results
9. [x] Verify the CSV contains the same set of filtered items

As a user without the administrator or content admin roles

1. [x] Visit /admin/content/audit/vamc-system-service-audit and verify you do not have access

### CODE REVIEW

**Review by kevwalsh — APPROVED** [review]
This is great stuff. 

I tested a bit with regex, which i've never used in Drupal views filters before

`<h3>[^C]` Revealed 39 instances of h3s that don't start with C
`<h3>[^Care]` Revealed 34 results of h3s that don't start with Care.
`<h4>` revealed 
 [This search](https://pr6733-6rhvouqlarqtfooxropsbkj365ilb4kn.ci.cms.va.gov/admin/content/audit/care-we-provide?title_op=contains&title=&field_body_value_op=not_starts&field_body_value=%3Ch3%3ECare+we+provide)(not Regex) revealed 85 results that don't _start_ with `<h3>Care we provide at`. Some of them start with `<h3 id="something">Care we provide`

In short, i think this audit will help us to 
1. Identify patterns that are not standard
2. Identify places where editors have gone away fromany pattern
3. Determine the scope of migration (how many variants our migration can handle)
4. Determine the scope of manual fixes
5. Determine the scope of change management with editors.

File: tests/behat/drupal-spec-tool/views.feature [hunk 1]
```diff
@@ -4,7 +4,7 @@ Feature: Views
   As a site owner
   I want to have views for various contexts and applications.
 
-  @dst @views
+  @dst @views
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/views.feature:7]: This leads me to believe you have some behat linter that is running and adding spaces to this.

File: config/sync/views.view.content.yml [hunk 2]
```diff
@@ -619,51 +610,1432 @@ display:
...
+        - url
+        - url.query_args
+        - user
+        - 'user.node_grants:view'
+        - user.roles
+      max-age: 0
+      tags:
+        - 'config:field.storage.node.field_administration'
+        - 'config:workflow_list'
+  care_audit_export:
```
Comment (swirtSJW) [inline config/sync/views.view.content.yml:894]: This seems too specific.   This report is specific to system services content not 'care'.

File: config/sync/views.view.content.yml [hunk 3]
```diff
@@ -619,51 +610,1432 @@ display:
...
+    display_plugin: data_export
+    id: care_audit_export
+    display_title: 'Care audit export'
```
Comment (swirtSJW) [inline config/sync/views.view.content.yml:897]: 'VAMC System service audit' would be a more honest pattern for what this is.

File: config/sync/views.view.content.yml [hunk 4]
```diff
@@ -619,51 +610,1432 @@ display:
...
+    position: 7
+    display_options:
+      display_extenders: {  }
+      path: admin/content/audit/care-we-provide/vacms-system-health-services-care-audit.csv
```
Comment (swirtSJW) [inline config/sync/views.view.content.yml:901]: update path to reflect VAMC System service audit

File: config/sync/views.view.content.yml [hunk 5]
```diff
@@ -619,51 +610,1432 @@ display:
...
+            content_admin: content_admin
+            administrator: administrator
+      title: 'VAMC System Health Services Care Audit'
+      relationships: {  }
+      pager:
+        type: none
+        options:
+          offset: 0
+      displays:
+        care_audit_page: care_audit_page
```
Comment (swirtSJW) [inline config/sync/views.view.content.yml:1523]: VAMC System service audit

File: config/sync/views.view.content.yml [hunk 6]
```diff
@@ -619,51 +610,1432 @@ display:
...
+        operator: AND
+        groups:
+          1: AND
+      access:
+        type: role
+        options:
+          role:
+            content_admin: content_admin
+            administrator: administrator
+      title: 'VAMC System Health Services Care Audit'
```
Comment (swirtSJW) [inline config/sync/views.view.content.yml:1516]:
```suggestion
      title: 'VAMC System Health Services Audit'
```

File: config/sync/views.view.content.yml [hunk 7]
```diff
@@ -619,51 +610,1432 @@ display:
...
+        - 'user.node_grants:view'
+        - user.roles
+      max-age: -1
+      tags:
+        - 'config:field.storage.node.field_administration'
+        - 'config:field.storage.node.field_body'
+    deleted: false
+  care_audit_page:
+    display_options:
+      path: admin/content/audit/care-we-provide
```
Comment (swirtSJW) [inline config/sync/views.view.content.yml:1584]: reflect VAMC System service audit

File: config/sync/views.view.content.yml [hunk 8]
```diff
@@ -619,51 +610,1432 @@ display:
...
+      menu:
+        type: tab
+        title: 'VAMC System Health Services Care Audit'
```
Comment (swirtSJW) [inline config/sync/views.view.content.yml:1587]:
```suggestion
        title: 'VAMC System Health Services Audit'
```

File: config/sync/views.view.content.yml [hunk 9]
```diff
@@ -816,90 +2199,57 @@ display:
...
-    display_title: Master
-    id: default
-    position: 0
+      access:
+        type: role
+        options:
+          role:
+            content_admin: content_admin
+            administrator: administrator
+      title: 'VAMC System Health Services Care Audit'
```
Comment (swirtSJW) [inline config/sync/views.view.content.yml:2213]:
```suggestion
      title: 'VAMC System Health Services Audit'
```

File: config/sync/views.view.content.yml [hunk 10]
```diff
@@ -816,90 +2199,57 @@ display:
...
+            items_per_page: false
+            items_per_page_label: 'Items per page'
+            items_per_page_options: '5, 10, 25, 50'
+            items_per_page_options_all: false
+            items_per_page_options_all_label: '- All -'
+            offset: false
+            offset_label: Offset
+          quantity: 9
+    display_plugin: page
+    display_title: 'Care audit'
```
Comment (swirtSJW) [inline config/sync/views.view.content.yml:2237]: VAMC System service audit

File: tests/behat/drupal-spec-tool/views.feature [hunk 11]
```diff
@@ -98,6 +98,8 @@ Feature: Views
 | Content | Bulk edit events | events_page | Page |
 | Content | Resources and support | resources_support_dashboard | Page |
 | Content | Resources and support landing page | resources_and_support_landing_page_block | Block |
+| Content | Care audit | care_audit_page | Page |
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/views.feature:101]: reflect VAMC System service audit

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Overall this looks good.  But there are bunch of places where the specific naming of the view is  misleading about what the view audits.   We can use it to audit access to care... but thankfully it is much more general and useful than that.   So the names should not mislead us about its abilities.

File: config/sync/views.view.content.yml (hunk 4 above)
Comment (JayDarnell) [reply config/sync/views.view.content.yml:901]: Updating

**Review by swirtSJW — APPROVED** [review]
This all QA's fine for me.
Idid find that there are no menu links for the new view nor an existing audit tool view.   I tried an easy config change that did not resolve it, so I made a new issue.
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/6759

**Comment by swirtSJW** [comment]
image
